core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -95,7 +98,15 @@ case class DeleteCommand(
       sparkSession.sharedState.cacheManager.recacheByPlan(sparkSession, target)
     }
 
-    Seq.empty[Row]
+    // Adjust for deletes at partition boundaries. Deletes at partition boundaries is a metadata
+    // operation, therefore we don't actually have any information around how many rows were deleted
+    // While this info may exist in the file statistics, it's not guaranteed that we have these
+    // statistics. To avoid any performance regressions, we currently just return a -1 in such cases
+    if (metrics("numRemovedFiles").value > 0 && metrics("numDeletedRows").value == 0) {
+      Seq(Row(-1L))
```

Comment:
I might be biased but I consider this type of code unsafe : a map is not a case class so the presence of a value cannot be guaranteed by the type system.

A safer alternative would be to make no assumption about the presence of those keys in the map, and explicitly handle their absence using optional values 

`metrics.get("numRemovedfiles").exists(_>0) && metrics.get("numDeletedRows").exists(_==0)`

Reply:
Hi @edmondo1984 I feel like this is pretty safe since we defined our metrics above in `createMetrics`, all within the same file. If someone ever deletes `numDeletedRows` from that metrics definition, our tests will catch it.

However, If you feel strongly about this, I can implement your suggestion.

Reply:
> Hi @edmondo1984 I feel like this is pretty safe since we defined our metrics above in `createMetrics`, all within the same file. If someone ever deletes `numDeletedRows` from that metrics definition, our tests will catch it.
> 
> However, If you feel strongly about this, I can implement your suggestion.

In term of concrete risks, you know the codebase better than I do, what's the risk that the metrics initialization code  gets moved somewhere else, that the lack of tests that confirm its content exposes to a bug, so if you are fine with it I have no reason to doubt it.

In terms of encouraging safer practices, I think if leaning towards the safer side does no harm, then it should be preferred

Reply:
Our tests would fail in this scenario because we'd at least see a key look up error